packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -249,44 +279,45 @@ Future<Null> findAndReplaceVersionProperties({@required BuildInfo buildInfo}) {
         .map((String line) {
 
           // find and replace build number
-          if (buildInfo.buildNumber != null) {
-            if (line.contains(new RegExp(r'^[ |\t]*(versionCode)[ =\t]*\d*'))) {
-              return line.splitMapJoin(new RegExp(r'(versionCode)[ =\t]*\d*'), onMatch: (Match m) {
-                return 'versionCode ${buildInfo.buildNumber}';
-              });
-            }
+          if (line.contains(_findVersionCode)) {
+            return line.splitMapJoin(_replaceVersionCode, onMatch: (Match m) {
+              return 'versionCode ${flutterManifest.buildNumber}';
+            });
           }
 
           // find and replace build name
-          if (buildInfo.buildName != null) {
-            if (line.contains(new RegExp(r'^[ |\t]*(versionName)[ =\t]*\"[0-9.]*"'))) {
-              return line.splitMapJoin(new RegExp(r'(versionName)[ =\t]*\"[0-9.]*"'), onMatch: (Match m) {
-                return 'versionName "${buildInfo.buildName}"';
-              });
-            }
+          if (line.contains(_findVersionName)) {
+            return line.splitMapJoin(_replaceVersionName, onMatch: (Match m) {
+              return 'versionName "${flutterManifest.buildName}"';
+            });
           }
           return line;
         })
-        .listen((String line) {
-            sink.writeln(line);
-          },
-          onDone: () {
-            sink.close();
+        .listen((String line) => sink.writeln(line),
+          onDone: () async {
```

Comment:
The `onDone` callback won't by awaited by default (its signature returns `void`).  You'll need to create a `Completer<void>` to signal when it finishes, then await that completer's future below.